minerl/herobraine/hero/handlers/agent/observations/inventory.py

```diff
@@ -93,35 +96,29 @@ def from_hero(self, obs):
             assert stack["quantity"] >= 0
             assert stack["metadata"] in range(16)
             if key is not None:
+                if key not in self.items:
+                    key = self._other
                 item_dict[key] += stack["quantity"]
 
         return item_dict
 
     def from_universal(self, obs):
-        item_dict = self.space.no_op()
-
+        item_dict = {item_id: 0 for item_id in self.items}  # Much faster than Dict.no_op()
         try:
             slots = _univ_obs_get_all_inventory_slots(obs)
 
             # Add from all slots
             for stack in slots:
-                if len(stack.keys()) == 0:
-                    # Skip this slot to maintain same behavior as pre-variant/metadata code.
-                    #
-                    # But really, this empty dict represents an "air" slot, and the
-                    # "air" if-statement a few lines down is never triggered.
-                    # See https://gist.github.com/shwang/3c3a5431f868a73e288d4a828e50d1c3
-                    # for an example univ.json, where all of the "air" slots are empty dicts
-                    # instead of dicts where `slot["name"] == "air"`.
-                    continue
-                item_type = mc.strip_item_prefix(stack['name'])
-
-                if item_type == "air" and "air" in item_dict:
-                    item_dict["air"] += 1
+                item_type = mc.strip_item_prefix(stack['name']) if len(stack.keys()) != 0 else "air"
+
+                if item_type == "air" and item_type in self.items:
+                    item_dict["air"] += 1  # This lets us count empty slots -non default MC behavior
```

Comment:
The change to counting empty (i.e. air) slots seems fine to me -- the reason the tests are failing right now is probably because they are asserting different "air" counts now that the counting logic is changed.

Could you look into that (using the test logs)?

https://github.com/minerllabs/minerl/blob/e8d16a790c2915a10aa6f36b03c74668704a5dc9/tests/unit/test_inventory_obs.py#L148-L156

Reply:
If there is something confusing about these tests, let me know and I can check them on my local machine

Reply:
The tests fail because we check if the obs.sample is in the space - but we expected a numpy object (with attribute .shape) where I was returning an int. I should have this fixed by returning the default dict of np.array(0)